pkg/solana/chainwriter/transform_registry.go

```diff
@@ -47,40 +56,71 @@ func CCIPExecuteArgsTransform(ctx context.Context, args any, accounts solana.Acc
 		return argsTransformed, accounts, nil
 	}
 
-	tokenPoolAddresses := []solana.PublicKey{}
-	for _, table := range registryTables {
-		tokenPoolAddresses = append(tokenPoolAddresses, table[0].PublicKey)
+	// Fetch all of the accounts in the pool lookup table with the proper IsWritable flag set
+	poolLookupAccounts, err := fetchPoolLookupAccounts(ctx, client, registryTables)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to fetch pool lookup accounts and set wrtiable flags: %w", err)
 	}
 
-	tokenIndexes := []uint8{}
-	for i, account := range accounts {
-		for _, address := range tokenPoolAddresses {
-			if account.PublicKey == address {
-				if i > 255 {
-					return nil, nil, fmt.Errorf("index %d out of range for uint8", i)
-				}
-				tokenIndexes = append(tokenIndexes, uint8(i)) //nolint:gosec
-			}
-		}
+	var aggregatedMessages []ccipocr3.Message
+	for _, report := range argsTransformed.Info.AbstractReports {
+		aggregatedMessages = append(aggregatedMessages, report.Messages...)
 	}
 
-	if len(tokenIndexes) != len(tokenPoolAddresses) {
-		return nil, nil, fmt.Errorf("missing token pools in accounts")
+	feeQuoterAddress, err := getFeeQuoterAddress(ctx, toAddress, args, tableMap, client)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to fetch fee quoter address: %w", err)
+	}
+
+	var tokenIndexes []uint8
+	// Accounts below are maintained to be in particular indexes in the Token Admin registry lookup table
+	poolProgram := poolLookupAccounts[2]
+	tokenProgram := poolLookupAccounts[6]
+	// Append token accounts to the account list and track at which index accounts for each token transfer starts
+	for _, message := range aggregatedMessages {
+		receiver := message.Receiver
+		var sourceChainSelector []byte
+		binary.LittleEndian.AppendUint64(sourceChainSelector, uint64(message.Header.SourceChainSelector))
+		for _, tokenAmount := range message.TokenAmounts {
+			destTokenAddress := tokenAmount.DestTokenAddress
+			userTokenAccount, err := getUserTokenAccount(receiver, tokenProgram.PublicKey, destTokenAddress)
+			if err != nil {
+				return nil, nil, fmt.Errorf("failed to calculate user token account PDA: %w", err)
+			}
+			perChainTokenConfig, err := getPerChainTokenConfig(sourceChainSelector, destTokenAddress, feeQuoterAddress)
+			if err != nil {
+				return nil, nil, fmt.Errorf("failed to calculate per chain per token config PDA: %w", err)
+			}
+			poolChainConfig, err := getPoolChainConfig(sourceChainSelector, destTokenAddress, poolProgram.PublicKey)
+			if err != nil {
+				return nil, nil, fmt.Errorf("failed to calculate pool chain config PDA: %w", err)
+			}
+			// Add token index at which token accounts will be appended
+			tokenIndexes = append(tokenIndexes, uint8(len(accounts))) //nolint:gosec
```

Comment:
This is incorrect, take a look at my original commit. The index is an offset from the start of "remaining accounts" (that includes the user specified accounts right before the tokens), so the value will be increasing.

Reply:
e.g. with a list like this:

```
... 14 mandatory accounts
<-- remaining_accounts starts here, this is `0`
4 user accounts for the program call
user token account 1 <- tokenIndexes for 1
per chain token config 1
pool chain config 1
...4 pool accounts 1 
user token account 2 <- tokenIndexes for 2
per chain token config 2
pool chain config 2
...3 pool accounts 2 
```

`tokenIndexes: would be [4, 11]`

Reply:
Oh right, missed [this](https://github.com/smartcontractkit/chainlink-ccip/blob/794d0c14d1bba6fe204178dac3c425beb689479e/chains/solana/contracts/programs/ccip-offramp/src/instructions/v1/execute.rs#L159). Think just updating to this should do the trick then
```suggestion
			tokenIndexes = append(tokenIndexes, uint8(len(accounts) - 14)) //nolint:gosec
```

Reply:
Pushed the fix in the latest commit